### ISSUE TITLE
Improve file downloads integration in a UWP app 

### ISSUE THREAD

**Issue by nirbil** [MicrosoftEdge/WebView2Feedback#2596]
Current Integration does not report downloaded file path.
Only a temporary downloaded file path and name is reported through the ResultFilePath variable.
After download completes the file is moved to its final destination but it is not reported by the integration where to.
The final path differs between Windows 10 and 11 and the file name itself can change according to preexisting files in the final folder.
I suggest fixing this by updating the ResultFilePath variable to the correct location upon DownloadOperation StateChange to Completed event.

[AB#40443951](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/40443951)

**Comment by champnic** [comment]
We've fixed this in runtimes 106.0.1355.0+. Thanks!

**Comment by nirbil** [comment]
Cool. Can you elaborate on the changes and how it should work after the fix? Is it like I suggested above?

**Comment by champnic** [comment]
Yes - ResultFilePath should now reflect the correct final location of the downloaded file when completed.

**Comment by nirbil** [comment]
I'm using the latest version (107.0.1379.1) and am not saying any changes. Upon download completion the ResultFilePath still points to the temporary app folder.

**Comment by champnic** [comment]
We'll take another look, thanks for checking.

**Comment by nirbil** [comment]
hey @champnic, any update on this?

**Comment by champnic** [comment]
Apologies I didn't update sooner - this is fixed in runtime versions 106.0.1355.0+. Are you still seeing this?

**Comment by nirbil** [comment]
Just retested it on current standard version (108.0.1462.46) and am still seeing the same behavior.

**Comment by champnic** [comment]
We'll try it again to verify the behavior.

**Comment by maurawinstanley** [comment]
I've verified the behavior and see the file path issue, we will investigate the bug and update the thread thanks.

**Comment by maurawinstanley** [comment]
@nirbil are you using a Windows 10 device?

**Comment by nirbil** [comment]
Yes I am. Did not check the behavior on Win11.

**Comment by maurawinstanley** [comment]
@nirbil the fix should be in canary on 2/1 if you want to try it out, thanks! Going to go ahead and close this out but feel free to re-open if the fix doesn't work for you.

**Comment by nirbil** [comment]
Looks great on latest canary, thanks!

**Comment by nirbil** [comment]
After further in depth testing there seem to be a couple of bugs with current implementation.
1. File naming is inconsistent. Sometimes the reported file name is not the correct one (in case there are duplicates then deletions), also if you remove and delete a downloaded file and later redownload it the file counter doesn't reset and you get an ever growing number suffix
2. There's a buildup of downloaded files in a folder within the app's data folder. I guess files are copied instead of moved to the download folder once download is finished. This probably explains some of the naming inconsistencies.

Should I open a new bug or reopen this one?

**Comment by champnic** [comment]
If you could open as a new issue that would be helpful, thanks!
